tests/sentry/models/test_apiapplication.py

```diff
@@ -13,15 +13,15 @@ def test_is_valid_redirect_uri(self):
         assert app.is_valid_redirect_uri("http://example.com/")
         assert app.is_valid_redirect_uri("http://example.com")
         assert app.is_valid_redirect_uri("http://example.com/.")
-        assert app.is_valid_redirect_uri("http://example.com//")
-        assert app.is_valid_redirect_uri("http://example.com/biz/baz")
+        assert not app.is_valid_redirect_uri("http://example.com//")
+        assert not app.is_valid_redirect_uri("http://example.com/biz/baz")
         assert not app.is_valid_redirect_uri("https://example.com/")
         assert not app.is_valid_redirect_uri("http://foo.com")
         assert not app.is_valid_redirect_uri("http://example.com.foo.com")
 
         assert app.is_valid_redirect_uri("http://sub.example.com/path")
-        assert app.is_valid_redirect_uri("http://sub.example.com/path/")
-        assert app.is_valid_redirect_uri("http://sub.example.com/path/bar")
+        assert not app.is_valid_redirect_uri("http://sub.example.com/path/")
```

Comment:
When it's not root of the domain, then the path with and without slash are considered different:
```
$ curl -v http://example.com/path 2>&1 | grep GET
> GET /path HTTP/1.1
$ curl -v http://example.com/path/ 2>&1 | grep GET
> GET /path/ HTTP/1.1
```